internal/stack/stack_processor.go

```diff
@@ -9,51 +9,78 @@ import (
 	"io/ioutil"
 	"path"
 	"sort"
+	"strings"
+	"sync"
 )
 
 // ProcessYAMLConfigFiles takes a list of paths to YAML config files, processes and deep-merges all imports,
 // and returns a list of stack configs
-func ProcessYAMLConfigFiles(filePaths []string, processStackDeps bool, processComponentDeps bool) ([]string, error) {
-	var result []string
+func ProcessYAMLConfigFiles(filePaths []string, processStackDeps bool, processComponentDeps bool) ([]string, map[string]interface{}, error) {
+	count := len(filePaths)
+	listResult := make([]string, count)
+	mapResult := map[string]interface{}{}
+	var errorResult error
+	var wg sync.WaitGroup
+	wg.Add(count)
+	mu := &sync.Mutex{}
+
+	for i, filePath := range filePaths {
+		go func(i int, p string) {
+			defer wg.Done()
+
+			config, importsConfig, err := ProcessYAMLConfigFile(p, map[string]map[interface{}]interface{}{})
+			if err != nil {
+				errorResult = err
+				return
+			}
 
-	for _, p := range filePaths {
-		config, importsConfig, err := ProcessYAMLConfigFile(p, map[string]map[interface{}]interface{}{})
-		if err != nil {
-			return nil, err
-		}
+			var imports []string
+			for k := range importsConfig {
+				imports = append(imports, k)
+			}
 
-		var imports []string
-		for k := range importsConfig {
-			imports = append(imports, k)
-		}
+			uniqueImports := u.UniqueStrings(imports)
+			sort.Strings(uniqueImports)
 
-		uniqueImports := u.UniqueStrings(imports)
-		sort.Strings(uniqueImports)
+			componentStackMap := map[string]map[string][]string{}
+			if processStackDeps {
+				componentStackMap, err = CreateComponentStackMap(p)
+				if err != nil {
+					errorResult = err
+					return
+				}
+			}
 
-		componentStackMap := map[string]map[string][]string{}
-		if processStackDeps {
-			componentStackMap, err = createComponentStackMap(p)
+			finalConfig, err := ProcessConfig(p, config, processStackDeps, processComponentDeps, "", componentStackMap, importsConfig)
 			if err != nil {
-				return nil, err
+				errorResult = err
+				return
 			}
-		}
 
-		finalConfig, err := ProcessConfig(p, config, processStackDeps, processComponentDeps, "", componentStackMap, importsConfig)
-		if err != nil {
-			return nil, err
-		}
+			finalConfig["imports"] = uniqueImports
 
-		finalConfig["imports"] = uniqueImports
+			yamlConfig, err := yaml.Marshal(finalConfig)
+			if err != nil {
+				errorResult = err
+				return
+			}
 
-		yamlConfig, err := yaml.Marshal(finalConfig)
-		if err != nil {
-			return nil, err
-		}
+			stackName := strings.TrimSuffix(strings.TrimSuffix(path.Base(p), ".yaml"), ".yml")
+
+			mu.Lock()
+			defer mu.Unlock()
```

Comment:
might want to consider what happens if this lock can't be obtained within some reasonable period of time?

Reply:
good call, let me look if something can be done w/o spending a lot of time on research